client/js/main.ts

```diff
@@ -201,59 +259,113 @@ function loadAttendees (filter: string = queryField.value, checkedIn: string = c
+	let query: string = `{
+		search_user(search: "${filter || ""}", n: 25, offset: 0, ${filterStr}) {
+			user {
+				id 
+				name 
+				email
+				questions(names: [${checked.join(",")}]) {
+					name
+					value
+				} 
+			} 
+			tags {
+				tag {
+					name 
+				} 
+				checked_in
+				checked_in_by
+				checked_in_date 
+			} 
+		} 
+	}`;
+
+	qwest.post("/graphql", JSON.stringify({
+		query: query
+	}), graphqlOptions).then((xhr, response: ISearchUserResponse) => {
+		let attendees: IGraphqlAttendee[] = response.data.search_user;
+
 		let attendeeList = document.getElementById("attendees")!;
 		let attendeeTemplate = <HTMLTemplateElement> document.getElementById("attendee-item")!;
 		let numberOfExistingNodes = document.querySelectorAll("#attendees li").length;
-		if (!attendeeList.firstChild || numberOfExistingNodes < response.length) {
+
+		if (!attendeeList.firstChild || numberOfExistingNodes < attendees.length) {
 			// First load, preallocate children
 			status.textContent = "Preallocating nodes...";
-			for (let i = numberOfExistingNodes; i < response.length; i++) {
+			for (let i = numberOfExistingNodes; i < attendees.length; i++) {
 				let node = document.importNode(attendeeTemplate.content, true) as DocumentFragment;
 				node.querySelector("li")!.style.display = "none";
 				node.querySelector(".actions > button")!.addEventListener("click", checkIn);
 				attendeeList.appendChild(node);
 			}
 			(<any> window).mdc.autoInit();
-			console.warn(`Allocated ${response.length - numberOfExistingNodes} nodes due to insufficient number`);
+			console.warn(`Allocated ${attendees.length - numberOfExistingNodes} nodes due to insufficient number`);
 			status.textContent = "Loading...";
 		}
 
 		// Reuse nodes already loaded from template
 		let existingNodes = document.querySelectorAll("#attendees li") as NodeListOf<HTMLElement>;
 		for (let i = 0; i < existingNodes.length; i++) {
-			let attendee = response[i];
+			let attendee = attendees[i];
 			if (!!attendee) {
 				existingNodes[i].style.display = "";
 
-				existingNodes[i].id = "item-" + attendee.id;
-				existingNodes[i].querySelector("#name")!.textContent = attendee.name;
-
-				let emails = attendee.emails.reduce((prev, current) => {
-					if (prev.indexOf(current) === -1) {
-						prev.push(current);
-					}
-					return prev;
-				}, <string[]> []);
-				existingNodes[i].querySelector("#emails")!.textContent = emails.join(", ");
+				existingNodes[i].id = "item-" + attendee.user.id;
+				existingNodes[i].querySelector("#name")!.textContent = attendee.user.name;
+				existingNodes[i].querySelector("#emails")!.textContent = attendee.user.email;
 
 				let button = existingNodes[i].querySelector(".actions > button")!;
 				let status = existingNodes[i].querySelector(".actions > span.status")!;
-				let date = attendee.tags[tag].checked_in_date;
-				if (date) {
+
+				// Determine if user has the current tag
+				let tagInfo: IGraphqlTag[] = attendee.tags.filter((curr) => {
```

Comment:
Can be simplified to `.filter(curr => curr.tag.name === tag)`. Also make sure you're using `===` and never `==`